src/avram/save_operation_template.cr

```diff
@@ -8,13 +8,21 @@ class Avram::SaveOperationTemplate
 
     class ::{{ type }}::SaveOperation < Avram::SaveOperation({{ type }})
       {% if primary_key_type.id == UUID.id %}
-        before_save set_uuid
-
-        def set_uuid
-          id.value ||= UUID.random()
-        end
+        # `before_save` calls `previous_def` which will not
+        # run this call on subclasses.
+        before_save set_primary_key_uuid
```

Comment:
I kept this here because this class isn't abstract, so you can still call `LineItem::SaveOperation.create` which would fail without this.